test/performance/odp_crc.c

```diff
@@ -0,0 +1,238 @@
+/* Copyright (c) 2021, Nokia
+ *
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier:     BSD-3-Clause
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <stdint.h>
+#include <inttypes.h>
+#include <getopt.h>
+
+#include <odp_api.h>
+#include <odp/helper/odph_api.h>
+
+/* Command line options */
+typedef struct {
+	uint32_t time;
+} options_t;
+
+static options_t options;
+static const options_t options_def = {
+	.time = 1000,
+};
+
+static void print_usage(void)
+{
+	printf("\n"
+	       "CRC performance test\n"
+	       "\n"
+	       "Usage: odp_crc_perf [options]\n"
+	       "\n"
+	       "  -t, --time    Approximate time (ms) for each measurement (default %u)\n"
+	       "  -h, --help    This help\n"
+	       "\n",
+	       options_def.time);
+}
+
+static int parse_options(int argc, char *argv[])
+{
+	int opt;
+	int long_index;
+	int ret = 0;
+
+	static const struct option longopts[] = {
+		{ "time", required_argument, NULL, 't' },
+		{ "help", no_argument, NULL, 'h' },
+		{ NULL, 0, NULL, 0 }
+	};
+
+	static const char *shortopts = "+t:h";
+
+	options = options_def;
+
+	while (1) {
+		opt = getopt_long(argc, argv, shortopts, longopts, &long_index);
+
+		if (opt == -1)
+			break;
+
+		switch (opt) {
+		case 't':
+			options.time = atol(optarg);
+			break;
+		case 'h':
+			/* fall through */
+		default:
+			print_usage();
+			ret = -1;
+			break;
+		}
+	}
+
+	if (options.time < 1) {
+		ODPH_ERR("Invalid time: %" PRIu32 "\n", options.time);
+		return -1;
+	}
+
+	return ret;
+}
+
+static uint8_t *buf, *data;
+static const int bufsize = 32 * 1024;
```

Comment:
`buf` and `data` could be normal function arguments instead of global variables.

I'd also prefer `bufsize` being either a define or an application option.

Reply:
In v2 we no longer test multiple buffer sizes, but this is simply a command line parameter, and the default is 32K. I hope this is what you meant.